### PR TITLE
RPN-891: supporting start & end in the orders

### PR DESCRIPTION
**Pull request by mystdeim**

### CODE REVIEW

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/model/Order.kt [hunk 1]
```diff
@@ -167,6 +171,13 @@ data class Order(
             return minOf(make, roundedMakeBalance)
         }
 
+        private fun active(start: Long?, end: Long?): Boolean {
```
Comment (yakischik) [inline order/core/src/main/kotlin/com/rarible/protocol/order/core/model/Order.kt:174]: Maybe isAlive?

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt [hunk 2]
```diff
@@ -107,6 +110,29 @@ class MongoOrderRepository(
...
     }
 
+    fun resetMakeStockAfter(now: Instant): Mono<UpdateResult> {
+        val query = Query(
+            Criteria().andOperator(
+                Order::end exists true,
+                Order::end lt now.toEpochMilli()
+            )
+        )
+        return template.updateFirst(query, Update().set("makeStock", EthUInt256.ZERO), Order::class.java)
```
Comment (yakischik) [inline order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt:120]: Order::makeStock.name is preferable than literals

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/job/OrderRecalculateMakeStockJob.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    private val orderUpdateService: OrderUpdateService
+) {
+    private val logger: Logger = LoggerFactory.getLogger(OrderRecalculateMakeStockJob::class.java)
+    private val orderRepository = MongoOrderRepository(reactiveMongoTemplate)
+
+    @Scheduled(initialDelay = 60000, fixedDelayString = "\${listener.resetMakeStockScheduleRate}")
+    fun update() = runBlocking {
+        if (properties.resetMakeStockEnabled.not()) return@runBlocking
+
+        logger.info("Starting update...")
```
Comment (yakischik) [inline order/listener/src/main/kotlin/com/rarible/protocol/order/listener/job/OrderRecalculateMakeStockJob.kt:31]: In logs it is better to write more specific messages like "Starting to update makeStock for orders"

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/job/OrderRecalculateMakeStockJob.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+        val result = orderRepository.resetMakeStockAfter(Instant.now()).awaitFirstOrNull()
+        logger.info("Successfully reset makeStock for ${result?.modifiedCount} documents")
+
+        orderRepository.findActualZeroMakeStock(Instant.now()).collect {
+            orderUpdateService.updateMakeStock(hash = it.hash)
+        }
+
+        logger.info("Successfully finished update().")
```
Comment (yakischik) [inline order/listener/src/main/kotlin/com/rarible/protocol/order/listener/job/OrderRecalculateMakeStockJob.kt:40]: The same - a bit more detailed message

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt [hunk 5]
```diff
@@ -107,6 +110,29 @@ class MongoOrderRepository(
         return template.query<Order>().matching(queue).all().asFlow()
     }
 
+    fun resetMakeStockAfter(now: Instant): Mono<UpdateResult> {
```
Comment (yakischik) [inline order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt:113]: Do we have index for such query?

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt [hunk 6]
```diff
@@ -107,6 +110,29 @@ class MongoOrderRepository(
...
+        val query = Query(
+            Criteria().andOperator(
+                Order::end exists true,
+                Order::end lt now.toEpochMilli()
+            )
+        )
+        return template.updateFirst(query, Update().set("makeStock", EthUInt256.ZERO), Order::class.java)
+    }
+
+    fun findActualZeroMakeStock(now: Instant): Flow<Order> {
```
Comment (yakischik) [inline order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt:123]: Do we have index for such query?

**Review by yakischik — CHANGES_REQUESTED** [review]

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/model/Order.kt (hunk 1 above)
Comment (mystdeim) [reply order/core/src/main/kotlin/com/rarible/protocol/order/core/model/Order.kt]: renamed

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt (hunk 2 above)
Comment (mystdeim) [reply order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt]: fixed

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/job/OrderRecalculateMakeStockJob.kt (hunk 4 above)
Comment (mystdeim) [reply order/listener/src/main/kotlin/com/rarible/protocol/order/listener/job/OrderRecalculateMakeStockJob.kt]: fxied

File: order/listener/src/main/kotlin/com/rarible/protocol/order/listener/job/OrderRecalculateMakeStockJob.kt (hunk 3 above)
Comment (mystdeim) [reply order/listener/src/main/kotlin/com/rarible/protocol/order/listener/job/OrderRecalculateMakeStockJob.kt]: added more text

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt (hunk 5 above)
Comment (mystdeim) [reply order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt:113]: added index

File: order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt (hunk 6 above)
Comment (mystdeim) [reply order/core/src/main/kotlin/com/rarible/protocol/order/core/repository/order/MongoOrderRepository.kt:123]: added index

**Review by yakischik — APPROVED** [review]

**Pull request merged**